src/handlers/transfers/prepare.js

```diff
@@ -140,14 +142,32 @@ const savePreparedRequest = async ({ validationPassed, reasons, payload, isFx, f
 const definePositionParticipant = async ({ isFx, payload, determiningTransferCheckResult }) => {
   const cyrilResult = await createRemittanceEntity(isFx)
     .getPositionParticipant(payload, determiningTransferCheckResult)
-  const account = await Participant.getAccountByNameAndCurrency(
-    cyrilResult.participantName,
-    cyrilResult.currencyId,
-    Enum.Accounts.LedgerAccountType.POSITION
-  )
+
+  let messageKey
+  const [debtorFsp, creditorFsp] = isFx ? [payload.initiatingFsp, payload.counterPartyFsp] : [payload.payerFsp, payload.payeeFsp]
```

Comment:
I think this is true only when its a payer initiated send fxtransfer. This should be fine for now.